### DISCUSSION TITLE
Can I make a Tracked Vehicle in carla simulator?

### DISCUSSION THREAD

**Discussion by KimBooKun** [carla-simulator/carla#7667] [General]
I wanna make tracked vehicle model (like panzer) in carla simulator and check physical movement.

I found a tracked vehicle model's json file in project chrono.
and also, I found it is possible to connect carla and chrono by using 'enable_chrono_physics' method in carla documentation.
This method uses just 3 inputs about vehicle's model. (1 vehicle.json, 2. powertrain.json, 3. tire.json)

There is only one example about wheeled vehicle by using tire.json.
But I need to use json file about track assembly, not tire.json

I want to ask if it 's possible to implement a Tracked Vehicle in carla simulator?
Or, is there any other possible way to make a tracked vehicle model in carla?

Please help me. Thank you for your kind help in advance. :)

**Comment by xavisolesoft** [comment]
Ye is it possible but modifying the CARLA C++ code.
Please, takes as example the implementation of the C++ birdge between CARLA and Chrono:
https://github.com/carla-simulator/carla/blob/dev/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Vehicle/MovementComponents/ChronoMovementComponent.cpp#L0-L1
